### PR TITLE
[Batch Relayer V4] [3] Adding chained references support for approveVault

### LINKED ISSUE #1506: Add support for chainedReferences in vaultApprove()
This seems to be the only function that doesn't use them.

### PR DESCRIPTION
**Pull request by jubeira**
Closes #1506.

### CODE REVIEW

File: pkg/standalone-utils/test/BaseRelayerLibrary.test.ts [hunk 1]
```diff
@@ -295,4 +297,36 @@ describe('BaseRelayerLibrary', function () {
...
+  describe('approve vault', () => {
+    function itApprovesVault(approveAmount: BigNumberish, allowance: BigNumberish) {
+      it('approves vault to use tokens', async () => {
+        const tx = await relayerLibrary.approveVault(token.address, approveAmount);
+        expectEvent.inIndirectReceipt(
+          await tx.wait(),
+          new ethers.utils.Interface(['event Approval(address indexed owner, address indexed spender, uint256 value)']),
+          'Approval',
+          { owner: relayerLibrary.address, spender: vault.address, value: allowance }
+        );
```
Comment (nventuro) [inline pkg/standalone-utils/test/BaseRelayerLibrary.test.ts:310]:
```suggestion

        expectEvent.inIndirectReceipt(
          await tx.wait(),
          token.interface,
          'Approval',
          { owner: relayerLibrary.address, spender: vault.address, value: allowance }
        );
```

File: pkg/standalone-utils/test/BaseRelayerLibrary.test.ts [hunk 2]
```diff
@@ -295,4 +297,36 @@ describe('BaseRelayerLibrary', function () {
...
+      });
```
Comment (nventuro) [inline pkg/standalone-utils/test/BaseRelayerLibrary.test.ts:311]:
```suggestion
        expect(await token.allowance(relayerLibrary.address, vault.address)).to.equal(allowance);
      });
```

**Review by nventuro — APPROVED** [review]
Let's merge into `master` instead of the current target branch

**Comment by jubeira** [comment]
Thanks for the review! Comments addressed; waiting for CI.

**Pull request merged**